text/0000-result-expect.md

```diff
@@ -0,0 +1,40 @@
+- Feature Name: `result_expect`
+- Start Date: 2015-05-13
+- RFC PR: (leave this empty)
+- Rust Issue: (leave this empty)
+
+# Summary
+
+Add an `expect` method to the Result type, bounded to `E: Debug`
```

Comment:
To be clear:

a) I'm just giving my interpretation of the vague current conventions, not something that I have 110% support for.

b) The idea is not that you should avoid `unwrap`/`expect`, it's that they act as assertions that should only fail during testing/development, when a *developer* is the one encountering the error rather than a *user*. Therefore it's fine to use `Debug` instead of `Display`. So for "tests, examples, wild experiments," and new projects, having a lot of panics is not a problem, because in theory the people encountering these errors are either developers or relatively advanced users testing an experimental/beta version. (If `Debug` output is impossible even for a developer to understand, that's a whole separate problem.)

(This discussion reminds me of [a rant I read](https://blogs.gentoo.org/lu_zero/2015/03/27/again-on-assert) about enabling the `assert` macro in release builds of C applications a while back. Though the situation is not exactly analogous, since in this case the assertion is still "there" in the release code, we just don't expect it to ever trigger, regardless of how bad the inputs to the program are.)